awx/main/signals.py

```diff
@@ -72,40 +72,65 @@ def get_current_user_or_none():
     return u
 
 
-def emit_event_detail(serializer, relation, **kwargs):
+def emit_event_detail(cls, relation, **kwargs):
     instance = kwargs['instance']
     created = kwargs['created']
     if created:
-        event_serializer = serializer(instance)
+        url = ''
+        if isinstance(instance, JobEvent):
+            url = '/api/v2/job_events/{}'.format(instance.id)
+        if isinstance(instance, AdHocCommandEvent):
+            url = '/api/v2/ad_hoc_command_events/{}'.format(instance.id)
+        group = camelcase_to_underscore(cls.__name__) + 's'
+        timestamp = instance.created.isoformat()
         consumers.emit_channel_notification(
-            '-'.join([event_serializer.get_group_name(instance), str(getattr(instance, relation))]),
-            event_serializer.data
+            '-'.join([group, str(getattr(instance, relation))]),
+            {
+                'id': instance.id,
+                relation.replace('_id', ''): getattr(instance, relation),
+                'created': timestamp,
+                'modified': timestamp,
+                'group_name': group,
+                'url': url,
+                'stdout': instance.stdout,
+                'counter': instance.counter,
+                'uuid': instance.uuid,
+                'start_line': instance.start_line,
+                'end_line': instance.end_line,
+                'event': instance.event,
+                'failed': instance.failed,
+                'changed': instance.changed,
+                'event_level': getattr(instance, 'event_level', ''),
+                'play': getattr(instance, 'play', ''),
+                'role': getattr(instance, 'role', ''),
+                'task': getattr(instance, 'task', ''),
+            }
         )
 
 
 def emit_job_event_detail(sender, **kwargs):
     from awx.api import serializers
-    emit_event_detail(serializers.JobEventWebSocketSerializer, 'job_id', **kwargs)
+    emit_event_detail(JobEvent, 'job_id', **kwargs)
 
 
 def emit_ad_hoc_command_event_detail(sender, **kwargs):
     from awx.api import serializers
-    emit_event_detail(serializers.AdHocCommandEventWebSocketSerializer, 'ad_hoc_command_id', **kwargs)
+    emit_event_detail(AdHocCommandEvent, 'ad_hoc_command_id', **kwargs)
 
 
 def emit_project_update_event_detail(sender, **kwargs):
     from awx.api import serializers
-    emit_event_detail(serializers.ProjectUpdateEventWebSocketSerializer, 'project_update_id', **kwargs)
+    emit_event_detail(ProjectUpdateEvent, 'project_update_id', **kwargs)
 
 
 def emit_inventory_update_event_detail(sender, **kwargs):
     from awx.api import serializers
-    emit_event_detail(serializers.InventoryUpdateEventWebSocketSerializer, 'inventory_update_id', **kwargs)
+    emit_event_detail(InventoryUpdateEvent, 'inventory_update_id', **kwargs)
```

Comment:
Since you're removing this use of `InventoryUpdateEventWebSocketSerializer`, I think it can be deleted. Searching for it...

https://github.com/ansible/awx/search?q=InventoryUpdateEventWebSocketSerializer&unscoped_q=InventoryUpdateEventWebSocketSerializer

It is used in the replay events tool. That just means the tool is no longer accurate, and that should just be changed to this pattern. That leaves only the definition of the serializer, which can be deleted.